client-react/src/pages/app/deployment-center/container/DeploymentCenterContainerAcrDataLoader.tsx

```diff
@@ -501,24 +475,30 @@ const DeploymentCenterContainerAcrDataLoader: React.FC<DeploymentCenterFieldProp
 
   useEffect(() => {
     if (registryIdentifiers.current[formProps.values.acrLoginServer]) {
-      fetchRepositories(formProps.values.acrLoginServer);
-      setAcrResourceId();
+      if (!acrUseManagedIdentities) {
+        fetchRepositories(formProps.values.acrLoginServer);
+      } else {
+        setAcrResourceId(formProps.values.acrLoginServer);
+      }
     }
-  }, [formProps.values.acrLoginServer]);
+  }, [formProps.values.acrLoginServer, acrUseManagedIdentities]);
 
   useEffect(() => {
-    if (registryIdentifiers.current[formProps.values.acrLoginServer] && formProps.values.acrImage) {
+    if (formProps.values.acrImage && !acrUseManagedIdentities) {
       fetchTags(formProps.values.acrImage);
     }
-  }, [formProps.values.acrLoginServer, formProps.values.acrImage]);
+  }, [formProps.values.acrImage]);
```

Comment:
Add missing dependency (used in line 487).

```suggestion
  }, [formProps.values.acrImage, acrUseManagedIdentities]);
```

Reply:
I have acrUseManagedIdentities as a dependency in line 484 for fetchRepositories. The flow I was hoping for was if acrUseManagedIdentities changes and admin credentials are selected, it would re-fetch the repositories, which would re-fetch the tags. I chose not to add that as a dependency here because fetchTags would get called twice.